src/components/wizard/wizard-step.component.ts

```diff
@@ -14,6 +14,9 @@ export class WizardStepComponent {
     /** The text to be displayed in the wizard step tab. */
     @Input() header: string;
 
+    /** If set to true it will become disabled when the current step is invalid **/
+    @Input() disableNextWhenInvalid: boolean | undefined = undefined;
```

Comment:
I am thinking this might be better not as `undefined` but with a default value of `false`? By default all types are `nullable`/`undefined` in typescript so the `| undefined = undefined` is adding code that is doing the same as the browser will do by default anyway.

That way the value will always be a `boolean` and we maintain existing behavior, and can removed the undefined checks in the `isNextDisabled` function (and same in step component)

Reply:
Doesn't this prevent you from doing `[isNextDisabled]="false"` on the step to override the parent value of true?

Reply:
Ah yes, good point actually, scrap the default value of false, but you can still just leave the type as:

```typescript
@Input() disableNextWhenInvalid: boolean;
```

Reply:
Updated

Reply:
Thanks, apologies for the unnecessary change, my misunderstanding